assets/agenda/utils.ts

```diff
@@ -215,7 +215,9 @@ export function isWatched(item: IAgendaItem | ICoverage, userId?: IUser['_id']):
  * @return {Boolean}
  */
 export function isCoverageForExtraDay(coverage: ICoverage, dateString: string) {
-    return coverage.scheduled != null && formatDate(moment(coverage.scheduled)) === dateString;
+    return coverage.scheduled != null && (
+        formatDate(moment(coverage.scheduled)) === dateString ||
+        formatDate(moment.utc(coverage.scheduled)) === dateString);
```

Comment:
when is that needed? can you provide an example?

Reply:
Let's consider a scenario where we have an event with planning item and no coverages, and its `planning_date` is set to `"2024-03-19T18:30:00.000Z"`. When we push this event to Newshub, the `display_date` in the agenda will be set to the `planning_date` (referenced [here](https://github.com/superdesk/newsroom-core/blob/develop/newsroom/push.py#L584)). In the agenda list view, the events are grouped by this date, and this particular event falls under the date `"2024-03-19T18:30:00.000Z".`

Now, let's say we add coverage to this event with a scheduled time of `"2024-03-20T19:29:59.000Z".` When we update the event, the `display_date` in the agenda changes accordingly. However, when looking for coverages in the list view, it still refers to the previous planning date, so no coverages appear.

Upon refreshing the page, the grouping is recalculated, and now the event is grouped based on the new display date. item is moved according to the new `display_date` and consequently, the coverages becomes visible in the list view.

Interestingly, when comparing the planning date with the coverage scheduled time using the same date formatting function, such as `formatDate(moment("2024-03-19T18:30:00.000Z")) === formatDate(moment.utc(coverage.scheduled)`), it evaluates to true.

Reply:
in that dateString `2024-03-19T18:30:00.000Z` which is `Mar 20, 2024`, but coverage scheduled time is `2024-03-20T19:29:59.000Z `which is `Mar 21, 2024`

Reply:
Check out this one comment: - https://sofab.atlassian.net/browse/NHUB-494?focusedCommentId=128599

Reply:
I got your point. we have two options to fix this, first option is that if first coverage is added in the previous event, we have to refresh the page to recalculate the event listing based on the new display date (coverage.scheduled) which means refetching all items.
or 
we have to update this logic as
`formatDate(moment(coverage.scheduled)) === dateString || formatDate(moment(item.planning_date)) === dateString`

or first we changed this logic:- 
https://github.com/superdesk/newsroom-core/blob/develop/newsroom/push.py#L572
```.py
if action_type == "add":
    display_dates = get_display_dates(agenda["dates"], agenda["planning_items"])
else:
    display_dates = orig_agenda["display_dates"]
    display_dates += get_display_dates(agenda["dates"], agenda["planning_items"])

agenda["display_dates"] = display_dates
```
in client update logic as 
`formatDate(moment(coverage.scheduled)) === dateString || formatDate(moment(item.display_date[0])) === dateString`

Reply:
so if the data we get from server are missing something I would handle it there, otherwise if we need to refetch the event when the coverage was added/updated that's also fine